wrappers/samtools/remove_decoys/wrapper.py

```diff
@@ -0,0 +1,19 @@
+__author__ = "Delvin So"
+__copyright__ = "Copyright 2020, Delvin So"
+__license__ = "MIT"
+
+from os import path
+from snakemake.shell import shell
+
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+bam = snakemake.input[0]
+decoy = snakemake.input[1]
+
+rm_reads = snakemake.output[0]
+out_f = snakemake.output[1]
+
+
+shell = (
+    "samtools view {bam} -b -h -t {snakemake.threads} -o {rm_reads} -U {out_f} -L  {decoy}"
+    "samtools view -t {snakemake.threads} {out_f}| grep -v hs37d5 | grep -v NC_007605 | samtools view - -hb > {out_f}"
```

Comment:
Pretty clever way of handling the bam header :)
Can you try using samtools reheader? http://www.htslib.org/doc/samtools-reheader.html
It edits headers in place, so this step could be much faster and use way less bandwidth VS printing out the whole bam again.

Reply:
We decided not the use samtools reheader. While the first line removes all reads mapping to the decoy chromosomes, it doesn't remove the chromosomes from being mentioned in the mate reference name (MRNM) in non-decoy chromosome aligned reads.

Furthermore, just running grep against the contents of the bam file does not generate a bam file of the same size as the two commands for some reason. We have gone with this solution for now.